cardano-config/src/Cardano/Config/Types.hs

```diff
@@ -324,38 +352,27 @@ instance FromJSON ViewMode where
 --   which verbosity to the log output.
 data TraceOptions = TraceOptions
   { traceVerbosity :: !TracingVerbosity
+  , traceBlockFetchClient :: !Bool
```

Comment:
This is fragile. Currently if you add a tracer, you would need to make sure that 'NodeConfiguration`'s `FromJSON` instance "aligns" with `TraceOptions`.

Potential solution: Create a data type isomorphic to bool for each tracer. I'd prefer a more elegant solution to this.

Reply:
If you split this type, you should be able to use Generic FromJSON, and have a nicer config format. Something like this:

```haskell
data TraceOptions = TraceOptions
  { traceVerbosity :: !TracingVerbosity
  , traceEnable :: !TraceEnable
  }
data TraceEnable = TraceEnable
  { traceBlockFetchClient :: !Bool
  , traceBlockFetchDecisions :: !Bool
  , traceBlockFetchProtocol :: !Bool
  , ...
  }
```

Then the YAML file could look like:
```yaml
tracing:
  verbosity: brief
  enable:
  - BlockFetchClient: false
  - BlockFetchDecisions: false
  - BlockFetchProtocol: false
  - ...
```

Reply:
@rvl the plan so far seems to be having the tracers formatted in the yaml file as follows:
(https://github.com/input-output-hk/cardano-node/issues/443#issuecomment-569678510)

```
TraceBlockFetchClient:
    Enabled: True
    MinSeverity: Debug
    Verbosity: MaximumVerbosity
```
Splitting `TraceOptions` doesn't alleviate the potential to mix up of tracers if the `FromJSON` instance of `NodeConfiguration` and `TraceOptions` records aren't aligned.